sdk/containerregistry/container-registry/src/blob/containerRegistryBlobClient.ts

```diff
@@ -0,0 +1,400 @@
+
+    try {
+      await this.client.containerRegistryBlob.deleteBlob(
+        this.repositoryName,
+        digest,
+        updatedOptions
+      );
+    } catch (e) {
+      span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Upload a manifest for an OCI artifact.
+   *
+   * @param manifest - the manifest to upload. If a resettable stream (a factory function that returns a stream) is provided, it may be called multiple times. Each time the function is called, a fresh stream should be returned.
+   */
+  public async uploadManifest(
+    manifest: (() => NodeJS.ReadableStream) | NodeJS.ReadableStream | OciManifest,
+    options?: UploadManifestOptions
+  ): Promise<UploadManifestResult> {
+    const { span, updatedOptions } = createSpan(
+      "ContainerRegistryBlobClient-uploadManifest",
+      options
+    );
+
+    try {
+      let manifestBody: Buffer | NodeJS.ReadableStream;
+
+      if (isReadableStream(manifest)) {
+        manifestBody = await readStreamToEnd(manifest);
+      } else if (typeof manifest === "function") {
+        manifestBody = await readStreamToEnd(manifest());
+      } else {
+        const serialized = serializer.serialize(Mappers.OCIManifest, manifest);
+        manifestBody = Buffer.from(JSON.stringify(serialized));
+      }
+
+      const tagOrDigest = options?.tag ?? (await calculateDigest(manifestBody));
+
+      const { dockerContentDigest } = await this.client.containerRegistry.createManifest(
+        this.repositoryName,
+        tagOrDigest,
+        manifestBody,
+        { contentType: KnownManifestMediaType.OciManifestMediaType, ...updatedOptions }
+      );
+
+      if (!dockerContentDigest) {
+        throw new Error("Digest not provided");
+      }
+
+      return { digest: dockerContentDigest };
+    } catch (e) {
+      span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Downloads the manifest for an OCI artifact
+   *
+   * @param tagOrDigest - a tag or digest that identifies the artifact
+   * @returns - the downloaded manifest
+   */
+  public async downloadManifest(
+    tagOrDigest: string,
+    options?: DownloadManifestOptions
+  ): Promise<DownloadManifestResult> {
+    const { span, updatedOptions } = createSpan(
+      "ContainerRegistryBlobClient-downloadManifest",
+      options
+    );
+
+    try {
+      const rawResponse = await new Promise<FullOperationResponse>(async (resolve) => {
+        this.client.containerRegistry.getManifest(this.repositoryName, tagOrDigest, {
+          accept: KnownManifestMediaType.OciManifestMediaType,
+          ...updatedOptions,
+          onResponse: (rawResponseProvided, flatResponse, error) => {
+            updatedOptions?.onResponse?.(rawResponseProvided, flatResponse, error);
+            resolve(rawResponseProvided);
```

Comment:
an interesting quirk here is that it's possible for `onResponse` to be called multiple times, but this would only resolve to the first one, rather than the usual behavior of taking the last